tests/suites/constraints/constraints.sh

```diff
@@ -1,88 +1,25 @@
-  juju add-machine --constraints "cores=2 mem=2G"
-
-  wait_for_machine_agent_status "0" "started"
-  wait_for_machine_agent_status "1" "started"
-
-  echo "Ensure machine 0 has 2 cores"
-  machine0_hardware=$(juju machines --format json | jq -r '.["machines"]["0"]["hardware"]')
-  check_contains "${machine0_hardware}" "cores=2"
-
-  echo "Ensure machine 1 has 2 cores and 2G memory"
-  machine1_hardware=$(juju machines --format json | jq -r '.["machines"]["1"]["hardware"]')
-  check_contains "${machine1_hardware}" "cores=2"
-  check_contains "${machine1_hardware}" "mem=2048M"
-
-  destroy_model "constraints-lxd"
-}
-
-run_constraints_aws() {
-  # Echo out to ensure nice output to the test suite.
-  echo
-
-  file="${TEST_DIR}/constraints-aws.txt"
-
-  ensure "constraints-aws" "${file}"
-
-  echo "Deploy 3 machines with different constraints"
-  juju add-machine --constraints "root-disk=16G"
-  juju add-machine --constraints "cores=4 root-disk=16G"
-  juju add-machine --constraints "instance-type=t2.nano"
-
-  wait_for_machine_agent_status "0" "started"
-  wait_for_machine_agent_status "1" "started"
-  wait_for_machine_agent_status "2" "started"
-
-  echo "Ensure machine 0 has 16G root disk"
-  machine0_hardware=$(juju machines --format json | jq -r '.["machines"]["0"]["hardware"]')
-  machine0_rootdisk=$(echo "$machine0_hardware" | awk '{for(i=1;i<=NF;i++){if($i ~ /root-disk/){print $i}}}')
-  check_ge "${machine0_rootdisk}" "root-disk=16384M"
-
-  echo "Ensure machine 1 has 4 cores and 16G root disk"
-  machine1_hardware=$(juju machines --format json | jq -r '.["machines"]["1"]["hardware"]')
-  machine1_cores=$(echo "$machine1_hardware" | awk '{for(i=1;i<=NF;i++){if($i ~ /cores/){print $i}}}')
-  machine1_rootdisk=$(echo "$machine1_hardware" | awk '{for(i=1;i<=NF;i++){if($i ~ /root-disk/){print $i}}}')
-  check_ge "${machine1_cores}" "cores=4"
-  check_ge "${machine1_rootdisk}" "root-disk=16384M"
-
-	echo "Ensure machine 2 has t2.nano instance type"
-  machine2_constraints=$(juju machines --format json | jq -r '.["machines"]["2"]["constraints"]')
-  check_contains "${machine2_constraints}" "instance-type=t2.nano"
-
-  destroy_model "constraints-aws"
-}
-
-test_constraints_lxd() {
-  if [ "$(skip 'test_constraints_lxd')" ]; then
-  		echo "==> TEST SKIPPED: constraints lxd"
+test_constraints_common() {
+  if [ "$(skip 'test_constraints_common')" ]; then
+  		echo "==> TEST SKIPPED: constraints common tests"
   		return
   fi
 
   (
   		set_verbosity
-  		cd .. || exit
-  		run "run_constraints_lxd"
-  )
-}
-
-test_constraints_aws() {
-  if [ "$(skip 'test_constraints_aws')" ]; then
-  		echo "==> TEST SKIPPED: constraints aws"
-  		return
-  fi
 
-  (
-  		set_verbosity
   		cd .. || exit
-  		run "run_constraints_aws"
+
+  		case "${BOOTSTRAP_PROVIDER:-}" in
+      "lxd" | "lxd-remote" | "localhost")
+        run "run_constraints_lxd"
+        ;;
+      "aws" | "ec2")
+        export BOOTSTRAP_PROVIDER="manual"
```

Comment:
why set bootstrap provider to manual?

Reply:
This is copy-paste issues, that I didn't catch on local tests (because test only on lxd). Already fixed